### PR TITLE
Add explicit export in fuse definition

### PR DESCRIPTION
**Pull request by ojha-ravi**
- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] pass the Travis CI test?

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://fusejs.io/

## Changes

I added explicit export statement to follow standard format and to be available for both global and module import use cases.

* Added generic type parameters for Fuse
* Changed the Namespace to Standard format (Capitalized)
* Added Export Statements to work with System Js setup
* Added test to verify generic parameters

### CODE REVIEW

File: fuse/index.d.ts [hunk 1]
```diff
@@ -3,22 +3,20 @@
...
 }
 
-declare namespace fuse {
+declare namespace Fuse {
 	interface IFuseOptions {
 		id?: string;
 		caseSensitive?: boolean;
 		include?: string[];
 		shouldSort?: boolean;
-		searchFn?: any;
```
Comment (andy-ms) [inline fuse/index.d.ts]: Why was this removed?

File: fuse/index.d.ts [hunk 2]
```diff
@@ -27,4 +25,12 @@ declare namespace fuse {
 		threshold?: number;
 		maxPatternLength?: number;
 	}
+
+	interface Options<T> extends IFuseOptions {
```
Comment (andy-ms) [inline fuse/index.d.ts]: Would be better to just remove`IFuseOptions` and use `Options<T>` for everything.

File: fuse/index.d.ts (hunk 1 above)
Comment (rbojha) [inline fuse/index.d.ts]: @andy-ms , this was a mistake from my side, fixed now..
and here is the reference for change -> [link](https://github.com/krisk/Fuse#options)

**Comment by rbojha** [comment]
@andy-ms, thanks for reviewing this PR, i have made the changes suggested by you, can you please take a final look and give your feedback?

Thanks again.

**Comment by andy-ms** [comment]
Thanks!
